apps/aecontract/src/aect_dispatch.erl

```diff
@@ -138,10 +138,15 @@ run_common(#{  amount      := Value
                     GasUsed = Gas - GasLeft,
                     Out = aevm_eeevm_state:out(ResultState),
                     {create_call(GasUsed, revert, Out, [], Call), Trees};
+                {error, out_of_gas, ErrorState} ->
+                    %% If we ran out of gas in a recursive call there
+                    %% might still be gas held back by the caller.
+                    GasLeft = aevm_eeevm_state:gas(ErrorState),
```

Comment:
My concern with this is whether there is any code path exiting with out_of_gas but not updating the gas in  the VM state.

Reply:
[Example](https://github.com/aeternity/epoch/blob/820a1c2502d49de7624ff319a0146195e98c2c50/apps/aevm/src/aevm_eeevm_state.erl#L338).

Reply:
(Maybe it does though did not check all code paths.)